### ISSUE TITLE
[no-unsafe-call/no-unsafe-member-access] False positives for multiple rules related to 'any' typings

### ISSUE THREAD

**Issue by adickenscheidt** [typescript-eslint/typescript-eslint#3978]
- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [read the FAQ](https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/FAQ.md) and my problem is not listed.

**Repro**

When using our project configuration, we get false positives for a number of rules regarding 'any' typings. We spent two days already trying to figure it out but are on our wits' end.

I uploaded a minimal project using our eslint configuration on my own github page: https://github.com/adickenscheidt/eslint-issue

Run `npm lint` to receive the errors.

**Expected Result**

No rule violations for this project.

**Actual Result**

When running the linter, we get the following rule violations on the validator.service.ts:

```eslint-issue\projects\eslint-issue\src\app\validator.service.ts
   5:2   error  Unsafe call of an `any` typed value          @typescript-eslint/no-unsafe-call
  10:12  error  Unsafe member access .min on an `any` value  @typescript-eslint/no-unsafe-member-access
  10:12  error  Unsafe call of an `any` typed value          @typescript-eslint/no-unsafe-call
```

**Additional Info**

```
ng lint

Linting "eslint-issue"...
  typescript-eslint:typescript-estree:parser parserOptions.project (excluding ignored) matched projects: Set(2) {
  'd:\\eslint-issue\\projects\\eslint-issue\\tsconfig.app.json',
  'd:\\eslint-issue\\projects\\eslint-issue\\tsconfig.spec.json'
} +0ms
  typescript-eslint:typescript-estree:createProjectProgram Creating project program for: D:\eslint-issue\projects\eslint-issue\src\app\app-routing.module.ts +0ms
  typescript-eslint:typescript-estree:createWatchProgram File did not belong to any existing programs, moving to create/update. d:\eslint-issue\projects\eslint-issue\src\app\app-routing.module.ts +0ms  typescript-eslint:typescript-estree:createWatchProgram Creating watch program for d:\eslint-issue\projects\eslint-issue\tsconfig.app.json. +1ms
  typescript-eslint:typescript-estree:createWatchProgram Creating watch program for d:\eslint-issue\projects\eslint-issue\tsconfig.spec.json. +1s
  typescript-eslint:parser:parser Resolved libs from program: [ 'es2018', 'dom' ] +0ms
  typescript-eslint:typescript-estree:createIsolatedProgram Getting isolated program in TS mode for: D:\eslint-issue\node_modules\@angular\core\fesm2015\core.js +0ms
  typescript-eslint:typescript-estree:createIsolatedProgram Getting isolated program in TS mode for: D:\eslint-issue\node_modules\@angular\router\fesm2015\router.js +2s
  typescript-eslint:typescript-estree:parser parserOptions.project (excluding ignored) matched projects: Set(2) {
  'd:\\eslint-issue\\projects\\eslint-issue\\tsconfig.app.json',
  'd:\\eslint-issue\\projects\\eslint-issue\\tsconfig.spec.json'
} +4s
  typescript-eslint:typescript-estree:createProjectProgram Creating project program for: D:\eslint-issue\projects\eslint-issue\src\app\app.component.spec.ts +4s
  typescript-eslint:typescript-estree:createWatchProgram Found existing program for file. d:\eslint-issue\projects\eslint-issue\src\app\app.component.spec.ts +3s
  typescript-eslint:parser:parser Resolved libs from program: [ 'es2018', 'dom' ] +2s
  typescript-eslint:typescript-estree:createIsolatedProgram Getting isolated program in TS mode for: D:\eslint-issue\node_modules\@angular\core\testing.d.ts +376ms
  typescript-eslint:typescript-estree:createIsolatedProgram Getting isolated program in TS mode for: D:\eslint-issue\node_modules\@angular\core\testing\testing.d.ts +7ms
  typescript-eslint:typescript-estree:createIsolatedProgram Getting isolated program in TS mode for: D:\eslint-issue\node_modules\@angular\router\testing.d.ts +80ms
  typescript-eslint:typescript-estree:createIsolatedProgram Getting isolated program in TS mode for: D:\eslint-issue\node_modules\@angular\router\testing\testing.d.ts +6ms
  typescript-eslint:typescript-estree:createIsolatedProgram Getting isolated program in TS mode for: D:\eslint-issue\projects\eslint-issue\src\app\app.component.ts +18ms
  typescript-eslint:typescript-estree:parser parserOptions.project (excluding ignored) matched projects: Set(2) {
  'd:\\eslint-issue\\projects\\eslint-issue\\tsconfig.app.json',
  'd:\\eslint-issue\\projects\\eslint-issue\\tsconfig.spec.json'
} +190ms
  typescript-eslint:typescript-estree:createProjectProgram Creating project program for: D:\eslint-issue\projects\eslint-issue\src\app\app.component.ts +190ms
  typescript-eslint:typescript-estree:createWatchProgram File did not belong to any existing programs, moving to create/update. d:\eslint-issue\projects\eslint-issue\src\app\app.component.ts +190ms   
  typescript-eslint:parser:parser Resolved libs from program: [ 'es2018', 'dom' ] +190ms
  typescript-eslint:typescript-estree:parser parserOptions.project (excluding ignored) matched projects: Set(2) {
  'd:\\eslint-issue\\projects\\eslint-issue\\tsconfig.app.json',
  'd:\\eslint-issue\\projects\\eslint-issue\\tsconfig.spec.json'
} +15ms
  typescript-eslint:typescript-estree:createProjectProgram Creating project program for: D:\eslint-issue\projects\eslint-issue\src\app\app.module.ts +15ms
  typescript-eslint:typescript-estree:createWatchProgram File did not belong to any existing programs, moving to create/update. d:\eslint-issue\projects\eslint-issue\src\app\app.module.ts +16ms       
  typescript-eslint:typescript-estree:createWatchProgram File was not found in program - triggering folder update. d:\eslint-issue\projects\eslint-issue\src\app\app.module.ts +1ms
  typescript-eslint:typescript-estree:createWatchProgram File was still not found in program after directory update - checking file deletions. d:\eslint-issue\projects\eslint-issue\src\app\app.module.ts +181ms
  typescript-eslint:parser:parser Resolved libs from program: [ 'es2018', 'dom' ] +198ms
  typescript-eslint:typescript-estree:createIsolatedProgram Getting isolated program in TS mode for: D:\eslint-issue\node_modules\@angular\platform-browser\fesm2015\platform-browser.js +272ms
  typescript-eslint:typescript-estree:createIsolatedProgram Getting isolated program in TS mode for: D:\eslint-issue\node_modules\@ngrx\store\fesm2015\ngrx-store.js +106ms
  typescript-eslint:typescript-estree:createIsolatedProgram Getting isolated program in TS mode for: D:\eslint-issue\projects\eslint-issue\src\app\app-routing.module.ts +108ms
  typescript-eslint:typescript-estree:parser parserOptions.project (excluding ignored) matched projects: Set(2) {
  'd:\\eslint-issue\\projects\\eslint-issue\\tsconfig.app.json',
  'd:\\eslint-issue\\projects\\eslint-issue\\tsconfig.spec.json'
} +437ms
  typescript-eslint:typescript-estree:createProjectProgram Creating project program for: D:\eslint-issue\projects\eslint-issue\src\app\validator.service.ts +436ms
  typescript-eslint:typescript-estree:createWatchProgram File did not belong to any existing programs, moving to create/update. d:\eslint-issue\projects\eslint-issue\src\app\validator.service.ts +259ms
  typescript-eslint:typescript-estree:createWatchProgram File was not found in program - triggering folder update. d:\eslint-issue\projects\eslint-issue\src\app\validator.service.ts +300ms
  typescript-eslint:typescript-estree:createWatchProgram File was still not found in program after directory update - checking file deletions. d:\eslint-issue\projects\eslint-issue\src\app\validator.service.ts +282ms
  typescript-eslint:typescript-estree:createWatchProgram File was not found in program - triggering folder update. d:\eslint-issue\projects\eslint-issue\src\app\validator.service.ts +146ms
  typescript-eslint:typescript-estree:createWatchProgram File was still not found in program after directory update - checking file deletions. d:\eslint-issue\projects\eslint-issue\src\app\validator.service.ts +149ms
  typescript-eslint:typescript-estree:createDefaultProgram Getting default program for: D:\eslint-issue\projects\eslint-issue\src\app\validator.service.ts +0ms
  typescript-eslint:typescript-estree:createIsolatedProgram Getting isolated program in TS mode for: D:\eslint-issue\projects\eslint-issue\src\app\validator.service.ts +910ms
  typescript-eslint:parser:parser Resolved libs from program: [ 'esnext.full' ] +1s
  typescript-eslint:typescript-estree:createIsolatedProgram Getting isolated program in TS mode for: D:\eslint-issue\node_modules\@angular\forms\fesm2015\forms.js +18ms
  typescript-eslint:eslint-plugin:utils:types Found an "error" any type +0ms
  typescript-eslint:eslint-plugin:utils:types Found an "error" any type +5ms
  typescript-eslint:eslint-plugin:utils:types Found an "error" any type +1ms
  typescript-eslint:eslint-plugin:utils:types Found an "error" any type +1ms
  typescript-eslint:typescript-estree:parser parserOptions.project (excluding ignored) matched projects: Set(2) {
  'd:\\eslint-issue\\projects\\eslint-issue\\tsconfig.app.json',
  'd:\\eslint-issue\\projects\\eslint-issue\\tsconfig.spec.json'
} +1s
  typescript-eslint:typescript-estree:createProjectProgram Creating project program for: D:\eslint-issue\projects\eslint-issue\src\environments\environment.prod.ts +1s
  typescript-eslint:typescript-estree:createWatchProgram File did not belong to any existing programs, moving to create/update. d:\eslint-issue\projects\eslint-issue\src\environments\environment.prod.ts +579ms
  typescript-eslint:typescript-estree:createWatchProgram File was not found in program - triggering folder update. d:\eslint-issue\projects\eslint-issue\src\environments\environment.prod.ts +5ms
  typescript-eslint:typescript-estree:createWatchProgram File was still not found in program after directory update - checking file deletions. d:\eslint-issue\projects\eslint-issue\src\environments\environment.prod.ts +277ms
  typescript-eslint:typescript-estree:createWatchProgram File was not found in program - triggering folder update. d:\eslint-issue\projects\eslint-issue\src\environments\environment.prod.ts +133ms
  typescript-eslint:typescript-estree:createWatchProgram File was still not found in program after directory update - checking file deletions. d:\eslint-issue\projects\eslint-issue\src\environments\environment.prod.ts +136ms
  typescript-eslint:typescript-estree:createDefaultProgram Getting default program for: D:\eslint-issue\projects\eslint-issue\src\environments\environment.prod.ts +1s
  typescript-eslint:typescript-estree:createIsolatedProgram Getting isolated program in TS mode for: D:\eslint-issue\projects\eslint-issue\src\environments\environment.prod.ts +1s
  typescript-eslint:parser:parser Resolved libs from program: [ 'esnext.full' ] +1s
  typescript-eslint:typescript-estree:parser parserOptions.project (excluding ignored) matched projects: Set(2) {
  'd:\\eslint-issue\\projects\\eslint-issue\\tsconfig.app.json',
  'd:\\eslint-issue\\projects\\eslint-issue\\tsconfig.spec.json'
} +846ms
  typescript-eslint:typescript-estree:createProjectProgram Creating project program for: D:\eslint-issue\projects\eslint-issue\src\environments\environment.ts +846ms
  typescript-eslint:typescript-estree:createWatchProgram File did not belong to any existing programs, moving to create/update. d:\eslint-issue\projects\eslint-issue\src\environments\environment.ts +274ms
  typescript-eslint:typescript-estree:createWatchProgram File was not found in program - triggering folder update. d:\eslint-issue\projects\eslint-issue\src\environments\environment.ts +10ms
  typescript-eslint:typescript-estree:createWatchProgram File was still not found in program after directory update - checking file deletions. d:\eslint-issue\projects\eslint-issue\src\environments\environment.ts +149ms
  typescript-eslint:parser:parser Resolved libs from program: [ 'es2018', 'dom' ] +432ms
  typescript-eslint:typescript-estree:parser parserOptions.project (excluding ignored) matched projects: Set(2) {
  'd:\\eslint-issue\\projects\\eslint-issue\\tsconfig.app.json',
  'd:\\eslint-issue\\projects\\eslint-issue\\tsconfig.spec.json'
} +429ms
  typescript-eslint:typescript-estree:createProjectProgram Creating project program for: D:\eslint-issue\projects\eslint-issue\src\main.ts +429ms
  typescript-eslint:typescript-estree:createWatchProgram Found existing program for file. d:\eslint-issue\projects\eslint-issue\src\main.ts +8ms
  typescript-eslint:parser:parser Resolved libs from program: [ 'es2018', 'dom' ] +263ms
  typescript-eslint:typescript-estree:createIsolatedProgram Getting isolated program in TS mode for: D:\eslint-issue\node_modules\@angular\platform-browser-dynamic\fesm2015\platform-browser-dynamic.js +707ms
  typescript-eslint:typescript-estree:createIsolatedProgram Getting isolated program in TS mode for: D:\eslint-issue\projects\eslint-issue\src\app\app.module.ts +32ms
  typescript-eslint:typescript-estree:createIsolatedProgram Getting isolated program in TS mode for: D:\eslint-issue\projects\eslint-issue\src\environments\environment.ts +4ms
  typescript-eslint:typescript-estree:parser parserOptions.project (excluding ignored) matched projects: Set(2) {
  'd:\\eslint-issue\\projects\\eslint-issue\\tsconfig.app.json',
  'd:\\eslint-issue\\projects\\eslint-issue\\tsconfig.spec.json'
} +330ms
  typescript-eslint:typescript-estree:createProjectProgram Creating project program for: D:\eslint-issue\projects\eslint-issue\src\polyfills.ts +330ms
  typescript-eslint:typescript-estree:createWatchProgram Found existing program for file. d:\eslint-issue\projects\eslint-issue\src\polyfills.ts +330ms
  typescript-eslint:parser:parser Resolved libs from program: [ 'es2018', 'dom' ] +81ms
  typescript-eslint:typescript-estree:parser parserOptions.project (excluding ignored) matched projects: Set(2) {
  'd:\\eslint-issue\\projects\\eslint-issue\\tsconfig.app.json',
  'd:\\eslint-issue\\projects\\eslint-issue\\tsconfig.spec.json'
} +21ms
  typescript-eslint:typescript-estree:createProjectProgram Creating project program for: D:\eslint-issue\projects\eslint-issue\src\test.ts +20ms
  typescript-eslint:typescript-estree:createWatchProgram Found existing program for file. d:\eslint-issue\projects\eslint-issue\src\test.ts +25ms
  typescript-eslint:parser:parser Resolved libs from program: [ 'es2018', 'dom' ] +20ms
  typescript-eslint:typescript-estree:createIsolatedProgram Getting isolated program in TS mode for: D:\eslint-issue\node_modules\@angular\platform-browser-dynamic\testing.d.ts +64ms
  typescript-eslint:typescript-estree:createIsolatedProgram Getting isolated program in TS mode for: D:\eslint-issue\node_modules\@angular\platform-browser-dynamic\testing\testing.d.ts +3ms

D:\eslint-issue\projects\eslint-issue\src\app\validator.service.ts
   5:2   error  Unsafe call of an `any` typed value          @typescript-eslint/no-unsafe-call
  10:12  error  Unsafe member access .min on an `any` value  @typescript-eslint/no-unsafe-member-access
  10:12  error  Unsafe call of an `any` typed value          @typescript-eslint/no-unsafe-call

✖ 3 problems (3 errors, 0 warnings)

Lint errors found in the listed files.
```

**Versions**

| package                            | 4.28.2 |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `4.28.2` |
| `@typescript-eslint/parser`        | `4.28.2` |
| `TypeScript`                       | `4.3.2` |
| `ESLint`                           | `7.26.0` |
| `node`                             | `16.5.0` |

**Comment by bradzacher** [comment]
cc @JamesHenry as it's an angualr project using the angular CLI.

---

From what I can see from your config - `validator.service.ts` isn't matched by any of the projects you pass to us.

- [This tsconfig](https://github.com/adickenscheidt/eslint-issue/blob/master/projects/eslint-issue/tsconfig.app.json#L12-L14) only matches `.d.ts` files.
- [This tsconfig](https://github.com/adickenscheidt/eslint-issue/blob/master/projects/eslint-issue/tsconfig.spec.json#L14-L17) only matches `.d.ts` and `.spec.ts` files.

[This tsconfig](https://github.com/adickenscheidt/eslint-issue/blob/master/tsconfig.json) is not used by our tooling due to your [eslint config](https://github.com/adickenscheidt/eslint-issue/blob/master/projects/eslint-issue/.eslintrc.json#L8-L15).

Additionally - due to [your eslint config](https://github.com/adickenscheidt/eslint-issue/blob/master/projects/eslint-issue/.eslintrc.json#L16) we attempt to create a "default" program for `validator.service.ts`. Which means we use whatever settings TS decides are the default for the file - for example, decorator support is turned off.

You need to fix your config so files that you are linting are included in a tsconfig.
For example - if I add `./**/*.service.ts` to your `tsconfig.app.json` then the lint runs correctly.
